src/boot/linux.c

```diff
@@ -16,95 +16,22 @@
-                const void *ctx, const EFI_DEVICE_PATH *device_path, const void *file_buffer, size_t file_size) {
-
-        const ValidationContext *payload = ASSERT_PTR(ctx);
-
-        if (device_path != payload->device_path)
-                return false;
-
-        /* Security arch (1) protocol does not provide a file buffer. Instead we are supposed to fetch the payload
-         * ourselves, which is not needed as we already have everything in memory and the device paths match. */
-        if (file_buffer && (file_buffer != payload->addr || file_size != payload->len))
-                return false;
-
-        return true;
-}
-
-static EFI_STATUS load_image(EFI_HANDLE parent, const void *source, size_t len, EFI_HANDLE *ret_image) {
-        assert(parent);
-        assert(source);
-        assert(ret_image);
-
-        /* We could pass a NULL device path, but it's nicer to provide something and it allows us to identify
-         * the loaded image from within the security hooks. */
-        struct {
-                VENDOR_DEVICE_PATH payload;
-                EFI_DEVICE_PATH end;
-        } _packed_ payload_device_path = {
-                .payload = {
-                        .Header = {
-                                .Type = MEDIA_DEVICE_PATH,
-                                .SubType = MEDIA_VENDOR_DP,
-                                .Length = sizeof(payload_device_path.payload),
-                        },
-                        .Guid = STUB_PAYLOAD_GUID,
-                },
-                .end = {
-                        .Type = END_DEVICE_PATH_TYPE,
-                        .SubType = END_ENTIRE_DEVICE_PATH_SUBTYPE,
-                        .Length = sizeof(payload_device_path.end),
-                },
-        };
-
-        /* We want to support unsigned kernel images as payload, which is safe to do under secure boot
-         * because it is embedded in this stub loader (and since it is already running it must be trusted). */
-        install_security_override(
-                        validate_payload,
-                        &(ValidationContext) {
-                                .addr = source,
-                                .len = len,
-                                .device_path = &payload_device_path.payload.Header,
-                        });
-
-        EFI_STATUS ret = BS->LoadImage(
-                        /*BootPolicy=*/false,
-                        parent,
-                        &payload_device_path.payload.Header,
-                        (void *) source,
-                        len,
-                        ret_image);
-
-        uninstall_security_override();
-
-        return ret;
-}
-
 EFI_STATUS linux_exec(
                 EFI_HANDLE parent,
                 const char16_t *cmdline,
                 const struct iovec *kernel,
                 const struct iovec *initrd) {
 
         size_t kernel_size_in_memory = 0;
-        uint32_t compat_address;
+        uint32_t compat_address, address;
```

Comment:
maybe rename these something with "entry point".


i.e. `compat_entry_point` and `efi_entry_point`?